decidim-core/lib/decidim/coauthorable.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+module Decidim
+  # This concern contains the logic related to collective or shared authorship.
+  #
+  # Coauthorable shares nearly the same object interface as Authorable but with some differences.
+  # - `authored_by?(user)` is exactly the same
+  # - `normalized_authors` is pluralized
+  #
+  # All coauthors, including the initial author, will share the same permissions.
+  module Coauthorable
+    extend ActiveSupport::Concern
+
+    included do
+      has_many :coauthorships, as: :coauthorable, class_name: "Decidim::Coauthorship", dependent: :destroy
+      has_many :authors, through: :coauthorships, class_name: "Decidim::User"
+      has_many :user_groups, as: :coauthorable, class_name: "Decidim::UserGroup", through: :coauthorships
+
+      scope :from_author, ->(author) { joins(:coauthorships).where("decidim_coauthorships.decidim_author_id": author.id) }
+      scope :from_user_group, ->(user_group) { joins(:coauthorships).where("decidim_coauthorships.decidim_user_group_id": user_group.id) }
+
+      # Checks whether the user is author of the given proposal, either directly
+      # authoring it or via a user group.
+      #
+      # user - the user to check for authorship
+      def authored_by?(user)
+        coauthorships.where(author: user).or(coauthorships.where(user_group: user.user_groups)).exists?
+      end
+
+      # Returns the identities for the authors, whether they are user groups or users.
+      #
+      # Returns an Array of User and/or UserGroups.
+      def identities
+        coauthorships.order(:created_at).collect(&:identity)
+      end
+
+      # Syntactic sugar to access first coauthor
+      def creator_author
+        authors.first
+      end
+
+      # Syntactic sugar to access first coauthor
+      def creator_user_group
+        user_groups.first
```

Comment:
This logic seems wrong, the first user_group coauthoring a proposal might not be the creator of the proposal (could be the 4th coauthor in order, but the first that is a user_group)

Reply:
The problem if the name of the method not what it returns as what I wanted was the first user group.
However as it is for testing purposes so I'm refactoring by removing this method and using a simple `#creator` that instead returns the `Coauthorship`.